hikari/impl/event_manager_base.py

```diff
@@ -151,14 +275,14 @@ def get_listeners(
     ) -> typing.Collection[event_manager.CallbackT[event_manager.EventT_co]]:
         if polymorphic:
             listeners: typing.List[event_manager.CallbackT[event_manager.EventT_co]] = []
-            for subscribed_event_type, subscribed_listeners in self._listeners.items():
-                if issubclass(subscribed_event_type, event_type):
-                    listeners += subscribed_listeners
+            for cls in event_type.dispatches():
+                listeners.extend(self._listeners[cls])
+
```

Comment:
This will raise `KeyError` as not all the event_types will be in the listeners:
```suggestion
            for cls in event_type.dispatches():
                if (items := self._listeners.get(cls)):
                    listeners.extend(items)

```